site/content/integrate/apps/functionality/external-webhooks/_index.md

```diff
@@ -7,87 +7,98 @@ aliases:
+App webhooks support an API key-like authentication method. A Mattermost server-generated `webhook_secret` is provided during App installation and is used by the implementer to validate a webhook request.
+
+There are two steps to enable this authentication method:
+
+1. Set the `remote_webhook_auth_type` property of the App manifest to `secret`.
+2. Define an `on_install` [call]({{<ref "/integrate/apps/structure/call">}}) in the App manifest with the `app` expand property set to `all`.
+   The `webhook_secret` property of the `app` context field in the call request contains the webhook secret.
 
 {{<note>}}
-The `OnInstall` call must have the `app` expand field set to `summary` for the generated secret key to be populated in the call request.
+The webhook secret should be stored by the App for authenticating incoming webhook requests. The App [key-value store]({{<ref "/integrate/apps/functionality/kv-store">}}) is suitable for storing the webhook secret. 
 {{</note>}}
 
-For example, an Golang App's [manifest]({{<ref "/integrate/apps/structure/manifest">}}) would define an `OnInstall` call to get the secret like this:
-
-```go
-appManifest = apps.Manifest{
-    AppID:       apps.AppID("hello-world"),
-    Version:     apps.AppVersion("0.1.0"),
-    HomepageURL: "https://my-site/my-repo",
-    DisplayName: "Hello, world!",
-    RequestedPermissions: apps.Permissions{
-        apps.PermissionActAsBot,
-        apps.PermissionRemoteWebhooks,
-    },
-    RequestedLocations: apps.Locations{
-        apps.LocationChannelHeader,
-        apps.LocationCommand,
-        apps.LocationPostMenu,
-    },
-    Deploy: apps.Deploy{
-        HTTP: &apps.HTTP{
-            RootURL: "http://my-site:4000",
-        },
-    },
-    OnInstall: &apps.Call{
-        Path: "/installed",
-        Expand: &apps.Expand{
-            App: apps.ExpandSummary,
-        },
-    },
+When this authentication method is enabled, webhook requests will contain a parameter named `secret` which should be verified against the known webhook secret. If the secrets don't match, the webhook request can be considered invalid.
+
+Using the example from the previous section, the webhook URL will look like this:
+
+`http://my-mattermost-server/plugins/com.mattermost.apps/apps/my-app/webhook/my-webhook-path?secret=xxxxxxxxxxxxxxxx`
+
+An App's [manifest]({{<ref "/integrate/apps/structure/manifest">}}) would define an `on_install` call to get the secret like this:
+
+```json
+{
+	"app_id": "hello-world",
+    "version": "0.1.0",
+	"display_name": "Hello, world!",
+	"icon": "icon.png",
+	"homepage_url": "https://my-site/my-repo",
+	"requested_permissions": [
+		"act_as_bot",
+		"remote_webhooks"
+	],
+	"requested_locations": [
+		"/channel_header",
+		"/command",
+		"/post_menu"
+	],
+	"http": {
+		"root_url": "http://localhost:4000"
+	},
+	"remote_webhook_auth_type": "secret",
+	"on_install": {
+        "path": "/installed",
+        "expand": {
+            "app": "all"
+        }
+	}
```

Comment:
The indentation looks off here

Reply:
How so? Can you provide a screenshot?

Reply:
I cannot reproduce that. Could you include a screenshot from your display showing this misalignment?

Reply:
On the snippet I'm commenting on here

<img width="1294" alt="image" src="https://user-images.githubusercontent.com/6913320/208490990-5ff08110-7944-4f59-a0fa-3b793263b007.png">

Reply:
Here's what I see:
![image](https://user-images.githubusercontent.com/18126609/208509609-f3bf93b3-2c33-4edb-8827-90515e38a5ca.png)

I think this could be a display issue in your browser. Is the alignment also off in the Cnanges tab of the PR if whitespace is ignored?

Reply:
I see what @mickmister is seeing here.

Reply:
Silly me. I removed this block completely a few commits ago. :) Nothing to see here!